src/Chainweb/Pact/Backend/RelationalCheckpointer.hs

```diff
@@ -71,7 +71,7 @@ initRelationalCheckpointer'
 initRelationalCheckpointer' bstate sqlenv loggr gasEnv = do
     let dbenv = BlockDbEnv sqlenv loggr
     db <- newMVar (BlockEnv dbenv bstate)
-    runBlockEnv db $ initSchema >> vacuumDb
+    runBlockEnv db $ initSchema >> vacuumDb -- TODO: remove?
```

Comment:
Do you think it might be worthwhile to consider putting on auto-vacuum?

Reply:
Not sure, I don't understand the details well enough. I think, we should do some more testing around that.

In past test runs `vacuumDb` was causing nodes to run out of memory and crash for large dbs. But maybe that was related to the savepoint nesting bug. So we should run more tests first.

Reply:
`VACUUM` makes a copy of the main DB to a temporary file and reloads - this probably copies unresolved snapshots as well, OOM'ing the node, since we were accumulating GB's of unresolved transactions. It's probably fine now that we've solved that problem if we can prove that the 2xdb size upper bound won't cause the node to run out of memory. `AUTOVACUUM` only removes stale pages, so it's probably not what we want on startup for a checkpointer.